src/redisraft.c

```diff
@@ -1296,6 +1296,108 @@ static int cmdRaftRandom(RedisModuleCtx *ctx,
     return REDISMODULE_OK;
 }
 
+static int cmdRaftScan(RedisModuleCtx *ctx,
+                         RedisModuleString **argv, int argc)
+{
+    if (argc < 3) {
+        RedisModule_WrongArity(ctx);
+        return REDISMODULE_OK;
+    }
+
+    char cursor[15] = {'0', 0};
+    char slots[REDIS_RAFT_HASH_SLOTS] = {0};
+    char * slot_str;
+
+    size_t str_len;
+    const char * str = RedisModule_StringPtrLen(argv[1], &str_len);
+
+
+    if (strncasecmp("slots", str, str_len) != 0) {
+        strncpy(cursor, str, str_len);
+        if (argc != 4) {
+            RedisModule_WrongArity(ctx);
+            return REDISMODULE_OK;
+        }
+
+        str = RedisModule_StringPtrLen(argv[2], &str_len);
+        if (strncasecmp("slots", str, str_len) != 0) {
+            RedisModule_ReplyWithError(ctx, "ERR unexpected argument");
+            return REDISMODULE_OK;
+        }
+        str = RedisModule_StringPtrLen(argv[3], &str_len);
+        slot_str = strndup(str, str_len);
+    } else {
+        if (argc != 3) {
+            RedisModule_WrongArity(ctx);
+            return REDISMODULE_OK;
+        }
+        str = RedisModule_StringPtrLen(argv[2], &str_len);
+        slot_str = strndup(str, str_len);
+    }
+
+    int ret = parseHashSlots(slots, slot_str);
+    free(slot_str);
```

Comment:
For memory allocation, it's better to use `RM_..` family functions just to track memory usage and get OOM error if we are lucky. 

So, I'd use `RM_Alloc()`, `RM_Free()` and `RM_Strdup()`.